### ISSUE TITLE
Add allocation-delay-until request in canonical jobspec?

### ISSUE THREAD

**Issue by dongahn** [flux-framework/rfc#55]
Migrating this question from PR #29. I think we need something like this to support "delay allocation until the specified time: e.g., `sbatch`'s `--begin` option.

> Does it make sense to change walltime key to time key and the time key expresses both the walltime requirement and the start time requirement?

**Comment by morrone** [comment]
I think that a set of wall-clock time constraints would be reasonable.  Perhaps:

* start - allocation must begin at this fixed time
* start-after - allocation can begin at any time after this fixed time

The former, "start", might be configurable to be capability-limited (for instance, only LC staff can use it for scheduling DATs...or maybe it the capability is more complex: anyone can use "start" for 50 or fewer nodes, but more than 50 nodes requires LC staff capability).

And maybe more complex time-based constraints in the future "I would like 3 hours between 9am and 5pm on a Weekday".  That might allow for some medium size interactive jobs that don't fit in the interactive pool, but also would not require administrative staff to get involved to schedule.

**Comment by dongahn** [comment]
Yes, I think being able to specify such constraints in the job spec would be a great start! 

On, the "start at this fixed time": 

Since this is a guarantee the scheduler will have to provide, it seems the scheduler needs to be advanced reservation-aware? Something to think about, as we map out our scheduling primitives in a new RFC...

**Comment by dongahn** [comment]
@morrone: does the current canonical jobspec have ways to specify start time requirement variants as described above? At this point, I believe I can schedule them if they are specified with our jobspc. (Even starting at a fixed time -- which folks typically call advanced reservation.)

**Comment by morrone** [comment]
@dongahn Yes.  A scheduler may support any key/value pairs it likes in the attributes->system area.  "duration" is just an example.

**Comment by SteVwonder** [comment]
>    start - allocation must begin at this fixed time
>    start-after - allocation can begin at any time after this fixed time


Based on the discussions in https://github.com/flux-framework/flux-core/discussions/3243, I believe we will be implementing the `start-after` using dependencies.

Exactly how we plan to handle `start` is still up in the air.  As @dongahn mentioned, it will require the scheduler to make an advanced reservation to guarantee the start time.
